src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -9,6 +9,11 @@ namespace System.Text.Json.Serialization
 {
     internal partial class JsonClassInfo
     {
+        private JsonPropertyInfo AddPolicyProperty(Type propertyType, JsonSerializerOptions options)
+        {
+            return AddProperty(propertyType, propertyInfo : null, classType : propertyType, options);
```

Comment:
Instead of re-using `AddProperty`, should we have overloads all the way down, including `CreateProperty` so that we don't have such an artifact? It becomes quite difficult to follow what the call graph is doing when we pass the same instance of a parameter down, twice.

Reply:
See my comment below.